code/game/objects/items/weapons/implant/implants/carrion/explosive_spider.dm

```diff
@@ -2,9 +2,32 @@
 	name = "explosive spider"
 	icon_state = "spiderling_explosive"
 	spider_price = 40
+	var/devastation_range = -1
+	var/heavy_range = 0
+	var/weak_range = 2
+	var/flash_range = 6
+	var/det_time = 20
 
 /obj/item/weapon/implant/carrion_spider/explosive/activate()
 	..()
-	visible_message(SPAN_DANGER("[src] explodes!"))
-	explosion(get_turf(src), -1, 0, 2, 6)
+	var/obj/item/weapon/implant/carrion_spider/explosive/biteszadusto = new/obj/item/weapon/implant/carrion_spider/explosive(get_turf(src))
+	if((istype(wearer, /mob/living/simple_animal) || istype(wearer, /mob/living/carbon)))
+		visible_message(SPAN_WARNING("[src] pops out of [wearer] and flashes brightly!"))
+	else
+		visible_message(SPAN_WARNING("[src] flashes brightly!"))
+	biteszadusto.set_light(3,3, COLOR_ORANGE)
+	spawn(det_time)
+		biteszadusto.prime()
```

Comment:
If people are fast enough they will be able to squash the spider before it explodes so you could do something like `biteszadusto?.prime()` to avoid a runtime.

Reply:
changed